pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -335,7 +348,7 @@ public TransportCnx getCnx() {
         return this.cnx;
     }
 
-    private static final class MessagePublishContext implements PublishContext, Runnable {
+    private static final class MessagePublishContext implements PublishContext, Runnable, Position {
```

Comment:
I wonder why  `MessagePublishContext` implements `Position`

Reply:
It's tricky, `pulsar-broker` depends on `managed-ledger`, so we can't use `MessagePublishContext` directly in  `managed-ledger`, but I need to get position info from this context.

Reply:
> OK. I see. Should we write some comments to explain this？

Sure.